### PR TITLE
Feature/Proj 82

### PR DESCRIPTION
**Pull request by Emilo067**
# Create New Password Component

## Description

This PR introduces a new `CreateNewPassword` component, which is part of the password recovery flow. The component renders a form that allows users to set a new password and confirm it. It uses the following technologies and libraries:

- `useCreateNewPassword` hook to manage form submission and control.
- `@photo-fiesta/ui-lib` components for form inputs, buttons, and typography.
- SCSS module for styling.

## Features

- **Form Fields**: Two password fields - `newPassword` and `confirmPassword`.
- **Form Validation**: Includes validation to ensure that both passwords match, and that they meet the required length (6-20 characters).
- **Responsive Styling**: Uses SCSS modules for responsive and reusable styles.
- **Button**: Full-width button for submitting the form.

## Changes

- Added a new `CreateNewPassword` component.
- Implemented form fields for new password creation.
- Added SCSS styles in `createNewPassword.module.scss`.
- Integrated the `useCreateNewPassword` hook for form handling.

## How to Test

1. Navigate to the password recovery flow.
2. Enter a new password in the first field.
3. Confirm the new password in the second field.
4. Submit the form using the "Create new password" button.
5. Ensure that validation works for password length and matching.

## Screenshot

Dekstop
Mobile

## Checklist

- [x] The form correctly handles input values and validation.
- [x] Styling has been verified for both desktop and mobile.
- [x] Password confirmation validation works as expected.

### CODE REVIEW

File: src/features/auth/ui/createNewPassword/createNewPassword.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+/**
+ * CreateNewPassword Component
+ *
+ * This component renders a form that allows the user to create a new password by providing
+ * a new password and its confirmation. It is used in the password recovery process.
+ *
+ * @component
+ * @example
+ * import { CreateNewPassword } from '@/features/auth/ui/createNewPassword/CreateNewPassword';
+ *
```
Comment (Lelik7777) [inline src/features/auth/ui/createNewPassword/createNewPassword.tsx]: это можно удалить из примера, по большому не несет информационной нагрузки

File: src/features/auth/ui/createNewPassword/useCreateNewPassword.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  const { control, handleSubmit } = useForm<FormValues>({
+    defaultValues: {
+      confirmPassword: '',
+      newPassword: '',
+    },
+    resolver: zodResolver(CreateNewPasswordSchema),
+  })
+
+  const onSubmit = (data: FormValues) => {
+    console.log('Form Submitted:', data)
```
Comment (Lelik7777) [inline src/features/auth/ui/createNewPassword/useCreateNewPassword.ts]: add TODO:add logic and remove console.log

File: src/features/auth/ui/createNewPassword/useCreateNewPassword.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      .string()
+      .min(6, 'Password must be at least 6 characters')
+      .max(20, 'Password must be no more than 20 characters')
+      .regex(
+        PASSWORD_REGEX,
+        'Password must contain 0-9, a-z, A-Z, ! " # $ % & \' ( ) * + , - . / : ; < = > ? @ [ \\ ] ^ _` { | } ~}'
+      ),
+  })
+  .refine(data => data.newPassword === data.confirmPassword, {
+    message: 'The password must match the new password',
```
Comment (Lelik7777) [inline src/features/auth/ui/createNewPassword/useCreateNewPassword.ts:27]: а где отобразиться это сообщение ,если пароли не совпадают?

Comment (Emilo067) [reply src/features/auth/ui/createNewPassword/useCreateNewPassword.ts:27]: в данной строке указано: path: ['confirmPassword'],
т.е. в поле под именем confirmPassword

File: src/features/auth/ui/createNewPassword/useCreateNewPassword.ts (hunk 2 above)
Comment (Emilo067) [reply src/features/auth/ui/createNewPassword/useCreateNewPassword.ts]: исправил

File: src/features/auth/ui/createNewPassword/createNewPassword.tsx (hunk 1 above)
Comment (Emilo067) [reply src/features/auth/ui/createNewPassword/createNewPassword.tsx]: исправил

**Review by EvgeniyaYakunina — APPROVED** [review]

**Pull request merged**